pkg/controller/subnet.go

```diff
@@ -112,6 +113,12 @@ func (c *Controller) enqueueUpdateSubnet(old, new interface{}) {
 				"gateway node changes from %s to %s ", oldSubnet.Spec.GatewayNode, newSubnet.Spec.GatewayNode)
 		}
 
+		needUpdateIPAM := !reflect.DeepEqual(oldSubnet.Spec.ExcludeIps, newSubnet.Spec.ExcludeIps) ||
+			oldSubnet.Spec.CIDRBlock != newSubnet.Spec.CIDRBlock ||
+			oldSubnet.Spec.Gateway != newSubnet.Spec.Gateway
+
+		key = strings.Join([]string{key, strconv.FormatBool(needUpdateIPAM)}, "#")
```

Comment:
We should not add more complex logical in the enqueue function. Can we check if we should update the ipam in `ipam.AddOrUpdateSubnet`

Reply:
fixed